docs/extending-dream2nix.md

```diff
@@ -0,0 +1,56 @@
+# Extending dream2nix with external translators, builders etc.
+
+`dream2nix` can be extended while you are `init`ializing it.
+This can be done in a few ways. For extending, you need to
+utilize the `config.extra` option of the dream2nix config.
```

Comment:
While you explain later that `# you can declare modules using functions here` and IIRC you have added a validation for this that prevents mistakes, it would be also useful to describe the function limitation here.

Something to the effect of:
> Unfortunately it is not possible to specify subsystem elements using functions directly in the config.
> This is caused by the requirement of the config to be serialisable, as only pure translators written
> in the Nix language can be called by passing the config directly and not all useful translators
> can be pure.
> 
> This limitation can be worked around by extracting the definition of a subsystem or subsystem
> element to a separate file and referring to it by path, which dream2nix will be then able to
> import directly.

At least that's my understanding of the issue, correct me if I'm wrong.